ci/Jenkinsfile.tests

```diff
@@ -79,6 +79,11 @@ pipeline {
         }
       }
     }
+    stage('Commit Checks') {
+      steps {
+        make('lint-commits')
+      }
+    }
```

Comment:
Here I'm adding the Jenkins job for linting the commits as the last stage. This is done so the rest of the tests can be evaluated and reported before checking for `donotmerge` since the usage of `donotmerge` is likely intentional, and we'd want to see the test failures first.